ts-rs/src/lib.rs

```diff
@@ -367,12 +367,7 @@ pub trait TS {
         struct Visit<'a>(&'a mut Vec<Dependency>);
         impl<'a> TypeVisitor for Visit<'a> {
             fn visit<T: TS + 'static + ?Sized>(&mut self) {
-                if T::transparent() {
-                    // the dependency `T` is "transparent", meaning that our original type depends
-                    // on the dependencies of `T` as well.
-                    T::dependency_types().for_each(self);
-                } else if let Some(dep) = Dependency::from_ty::<T>() {
-                    // the dependency `T` is not transparent, so we just add it to the output
+                if let Some(dep) = Dependency::from_ty::<T>() {
                     self.0.push(dep);
                 }
```

Comment:
I love that this just works. Awesome!

Reply:
Yeah, since `T::transparent` is always `false` we just get rid of that check